fuzz/src/chanmon_consistency.rs

```diff
@@ -403,11 +404,51 @@ pub fn do_test<Out: test_logger::Output>(data: &[u8], out: Out) {
 	}
 
 	loop {
+		macro_rules! check_send_res { ($send_err: expr) => { {
+			macro_rules! check_res { ($api_err: expr) => { {
+				match $api_err {
+					APIError::APIMisuseError { .. } => panic!("We can't misuse the API"),
+					APIError::FeeRateTooHigh { .. } => panic!("We can't send too much fee?"),
+					APIError::RouteError { .. } => panic!("Our routes should work"),
+					APIError::ChannelUnavailable { err } => {
+						// Test the error against a list of errors we can hit, and reject
+						// all others. If you hit this panic, the list of acceptable errors
+						// is probably just stale and you should add new messages here.
+						match err.as_str() {
+							"Peer for first hop currently disconnected/pending monitor update!" => test_return!(),
+							_ if err.starts_with("Cannot push more than their max accepted HTLCs ") => test_return!(),
+							_ if err.starts_with("Cannot send value that would put us over the max HTLC value in flight our peer will accept ") => test_return!(),
+							_ if err.starts_with("Cannot send value that would put our balance under counterparty-announced channel reserve value") => test_return!(),
+							_ if err.starts_with("Cannot send value that would overdraw remaining funds.") => test_return!(),
+							_ if err.starts_with("Cannot send value that would not leave enough to pay for fees.") => test_return!(),
+							_ => panic!(err),
+						}
```

Comment:
> Critically, we keep going after hitting an error, as there's no
> reason channels should get out of sync even if a send fails.

Doesn't this still result in the test returning if a send fails?

Reply:
No - we only `return` from `do_test` if either we read a byte we don't know how to process or we run out of bytes. Prior to this we would `test_return!()` if we max out sending over a channel as well.

Reply:
Aye, think it was changed as the quoted code is using `test_return!` but now is `{}`. So should be good.